lib/vrl/web-playground/public/index.html

```diff
@@ -88,83 +88,150 @@ <h1>[draft] VRL playground wasm</h1>
+          // we assume the user has passed in valid json on each
+          // line
+          if (lines[1][0] == "{" && lines[1][lines[1].length - 1] == "}") {
+            return true;
+          }
         }
+        return false;
       }
+      function handleRunCode(input) {
+        if (isJsonL()) {
+          let inputs = [];
+          let program = window.programEditor.getValue();
+          let lines = window.eventEditor.getModel().getLinesContent();
+          lines.forEach((line) => {
+            inputs.push({
+              program: program,
+              event: tryJsonParse(line)
+            })
+          });
+
+          let results = [];
+          inputs.forEach((input) => {
+            results.push(window.run_vrl(input));
+          })
+          let outputs = [];
+          results.forEach((res) => {
+            if (res.output) {
+              outputs.push(JSON.stringify(res["result"], null, "\t"));
+            } else if (res.msg) {
+              outputs.push(res["msg"]);
+            }
+          })
+          // disable output validation for json
+          // since jsonl input won't ouput valid json
+          monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+            validate: false
+          });
+          window.outputEditor.setValue(outputs.join("\n"));
+          return results;
+        }
+
+        if (input == null) {
+          input = {
+            program: window.programEditor.getValue(),
+            event: tryJsonParse(window.eventEditor.getValue()),
+          };
+        }
 
-      let res = window.run_vrl(input);
+        let res = window.run_vrl(input);
 
-      console.log("[DEBUG::handleRunCode()] Printing out res: ", res);
-      if (res.output) {
-        window.outputEditor.setValue(JSON.stringify(res["result"], null, "\t"));
-      } else if (res.msg) {
-        window.outputEditor.setValue(res["msg"]);
+        console.log("[DEBUG::handleRunCode()] Printing out res: ", res);
+        if (res.output) {
+          window.outputEditor.setValue(JSON.stringify(res["result"], null, "\t"));
+        } else if (res.msg) {
+          window.outputEditor.setValue(res["msg"]);
+        }
+        return res;
       }
-      return res;
-    }
-
-  </script>
-  <script>
-    function handleShareCode() {
-      let state = {
-        program: window.programEditor.getValue(),
-        event: JSON.parse(window.eventEditor.getValue())
+    </script>
+    <script>
+      function handleShareCode() {
+        let state = {
+          program: window.programEditor.getValue(),
+          event: JSON.parse(window.eventEditor.getValue()),
+        };
+
+        console.log("[DEBUG::handleShareCode()] Printing out state", state);
+        console.log(
+          "[DEBUG::handleShareCode()] Printing out base64 encoded state\n",
+          btoa(JSON.stringify(state))
+        );
+        window.history.pushState(state, "", `?state=${encodeURIComponent(btoa(JSON.stringify(state)))}`);
```

Comment:
this change fixes the bug we had to use encodeURIComponent when generating the sharable URL